### DISCUSSION TITLE
File based routing: how can I create a layout that wrap all the nested children?

### DISCUSSION THREAD

**Discussion by mattiaz9** [TanStack/router#1804] [Q&A]
I see that using the `_` prefix is used to selectively add layout to routes but that system can become quite repetitive.
Is it possible to have something like the `__root` layout for the inner routes that doesn't have to be repeated every time?

**Comment by schiller-manuel** [comment]
Layout routes should already wrap all of its children, so I don't understand the question.
Can please you add a complete minimal example by forking one of the examples on stackblitz?

**Reply by mattiaz9** [reply]
I'm working on a starter template, this is a fork: https://codesandbox.io/p/devbox/vigilant-dan-smqjmt

I used a layout named `_base` to add the navbar, but when I navigate to the blog section it disappear, so it didn't wrap that layout to its children.

The only way I found out to make it work is to be explicit and nesting all the files in a folder named `_base`.

**Reply by schiller-manuel** [reply]
> The only way I found out to make it work is to be explicit and nesting all the files in a folder named _base.

That's exactly how you do it, also see the docs: https://tanstack.com/router/latest/docs/framework/react/guide/routing-concepts#pathless--layout-routes

**Reply by schiller-manuel** [reply]
you could also rename the folder `blog` to `_base.blog`: https://codesandbox.io/p/devbox/long-sun-9f22q3
but that's just another way of expressing the same thing
